lib/puppet/provider/ec2_vpc_routetable/v2.rb

```diff
@@ -0,0 +1,113 @@
+require_relative '../../../puppet_x/puppetlabs/aws.rb'
+
+Puppet::Type.type(:ec2_vpc_routetable).provide(:v2, :parent => PuppetX::Puppetlabs::Aws) do
+  confine feature: :aws
+
+  mk_resource_methods
+  remove_method :tags=
+
+  def self.instances
+    regions.collect do |region|
+      response = ec2_client(region).describe_route_tables()
+      tables = []
+      response.data.route_tables.each do |table|
+        hash = route_table_to_hash(region, table)
+        tables << new(hash) if hash[:name]
+      end
+      tables
+    end.flatten
+  end
+
+  read_only(:region, :vpc, :routes)
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name] # rubocop:disable Lint/AssignmentInCondition
+        resource.provider = prov if resource[:region] == prov.region
+      end
+    end
+  end
+
+  def self.route_to_hash(region, route)
+    ec2 = ec2_client(region)
+    gateway = if route.gateway_id == 'local'
+      'local'
+    else
+      begin
+        igw_response = ec2.describe_internet_gateways(internet_gateway_ids: [route.gateway_id])
+        name_from_tag(igw_response.data.internet_gateways.first)
+      rescue Aws::EC2::Errors::InvalidInternetGatewayIDNotFound
+        nil
+      end
+    end
+    {
+      'destination_cidr_block' => route.destination_cidr_block,
```

Comment:
Unfortunately yes. See the same thing in the security group provider too.

For nested values (like this and the ingress rules) the comparison of the array within insync relies on these being strings, as Puppet automatically coerces the other side of the equality to strings.